MISRA.md

```diff
@@ -31,6 +31,35 @@ _Ref 8.9.1_
        order of execution, some variables have file scope definitions rather
        than function scope.
 
+#### Rule 8.13
+_Ref 8.13.1_
+
+- MISRA C-2012 Rule 8.13 Parameter passed is never used, should be declared as
+    const.  The argument passed to the `prvIPTask` function is left unused which is
+    considered as the variable not being used and thus warranting the use of `const`.
+    However, the FreeRTOS-kernel function `xTaskCreate` expects a function signature
+    of type `void vSomeFunction( void * pvArgs )`. To satisfy that requirement, the
+    function signature of `prvIPTask` does not have a `const` qualifier in the
+    parameter signature.
+
+#### Rule 10.5
+_Ref 10.5.1_
+
+- MISRA C-2012 Rule 10.5 Converting from an unsigned to an enum type. The
+    operation is safe to perform in that case, as we are using a generic API
+    to send and receive data, in that case the exact data sent it is received
+
+#### Rule 11.1
+_Ref 11.1.1_
+
+- MISRA C-2012 Rule 11.1 Converting from a void pointer to a function pointer.
+   The `FreeRTOS_setsockopt` API allows users to configure sockets by setting
+   various options. In order to do so, the function must accept one parameter
+   which, based on the option value, can be casted to the corresponding socket
+   field. To that end, that parameter is of `void *` type to accommodate all values.
+   The caller of the API is responsible for providing correct function pointer to the
+   API. Thus, this violation can be safely suppressed.
+
 #### Rule 11.3
 _Ref 11.3.1_
 
```

Comment:
What about for 64-bit targets that do not allow 32-bit aligned access?

Reply:
We currently do not support any hardware with that feature/drawback the only 64 bit we currently support is RiscV and they do support unaligned access (if i am not mistaken)
when we support more 64bit hardware we will have to add code to cover that case.

Reply:
The Cortex-A53 is only strict about alignment when configured such in the SCTLR. It's pretty common to configure it to allow unaligned access.

In contrast, a Cortex-M0 (ARMv6-M) will always fail unaligned accesses.. but is 4-byte aligned.

SPARC and MIPS are both example of a 64bit architectures that do not support unaligned access... So de-referencing a 32-bit aligned pointer as a 64bit type could cause a fault at runtime.

In the case of FR+TCP, we're unlikely to run into this problem in particular.